app/src/main/kotlin/com/hedvig/app/authenticate/UserViewModel.kt

```diff
@@ -89,39 +79,32 @@ class UserViewModel(
       .launchIn(viewModelScope)
   }
 
-  private suspend fun submitCode(grant: Grant) {
-    when (val result = authRepository.exchange(grant)) {
-      is AuthTokenResult.Error -> Timber.e(result.message)
+  private suspend fun submitCode(grant: AuthorizationCodeGrant) {
+    when (val authTokenResult = authRepository.exchange(grant)) {
+      is AuthTokenResult.Error -> e { authTokenResult.message }
       is AuthTokenResult.Success -> {
-        runCatching {
-          pushTokenManager.refreshToken()
-        }
-        onAuthSuccess(result.accessToken)
+        onAuthSuccess(authTokenResult)
       }
     }
   }
 
-  private suspend fun onAuthSuccess(accessToken: AccessToken) {
+  private suspend fun onAuthSuccess(authTokenResult: AuthTokenResult.Success) {
     hAnalytics.loggedIn()
     featureManager.invalidateExperiments()
-    authenticationTokenService.authenticationToken = accessToken.token
+    authTokenService.updateTokens(
+      authTokenResult.accessToken,
+      authTokenResult.refreshToken,
+    )
     uploadMarketAndLanguagePreferencesUseCase.invoke()
     mutableViewState.update {
       it.copy(navigateToLoggedIn = true)
     }
+    runCatching {
+      pushTokenManager.refreshToken()
+    }
```

Comment:
Hmm I added this since it was here already at some previous state of this file.
Maybe this should be at a different place, where login happens for all markets? Maybe should look into this if we haven't already and I've missed it.